queue_job/job.py

```diff
@@ -645,11 +668,45 @@ def eta(self, value):
         else:
             self._eta = value
 
+    def get_db_txid(self):
+        """
+        Get the current if of the database transaction the job is running in. Only store
+        the 32bit representation variant in the database so it can be compared directly
+        to the 'backend_xid' in pg_stat_activity.
+        """
+        if self.env.cr._cnx.server_version >= 130000:
+            self.env.cr.execute(
+                "SELECT (pg_current_xact_id()::text::bigint % (2^32)::bigint)::text"
+            )
+        else:
+            self.env.cr.execute("SELECT (txid_current() % (2^32)::bigint)::text;")
+        return self.env.cr.fetchone()[0]
+
+    def set_db_txid(self):
+        """
+        Use a new cursor to update the queue_job record of this job with the
+        transaction ID the process is running in. The new cursor is necessary because
+        after each commit the transaction ID changes (cfr. stored() is executed), which
+        makes the transaction ID useless otherwise.
+        """
```

Comment:
```suggestion
        """Store transaction ID on current job record.
        
        Use a new cursor to update the queue_job record of this job with the
        transaction ID the process is running in. The new cursor is necessary because
        after each commit the transaction ID changes (cfr. stored() is executed), which
        makes the transaction ID useless otherwise.
        """
```